src/panels/lovelace/cards/hui-tile-card.ts

```diff
@@ -0,0 +1,200 @@
+import { DOMAINS_TOGGLE, STATES_OFF } from "../../../common/const";
+import { computeDomain } from "../../../common/entity/compute_domain";
+import { computeStateDisplay } from "../../../common/entity/compute_state_display";
+import { stateIconPath } from "../../../common/entity/state_icon_path";
+import "../../../components/ha-card";
+import "../../../components/tile/ha-tile-icon";
+import "../../../components/tile/ha-tile-info";
+import { ActionHandlerEvent } from "../../../data/lovelace";
+import { HomeAssistant } from "../../../types";
+import { actionHandler } from "../common/directives/action-handler-directive";
+import { findEntities } from "../common/find-entities";
+import { handleAction } from "../common/handle-action";
+import { createEntityNotFoundWarning } from "../components/hui-warning";
+import { LovelaceCard, LovelaceCardEditor } from "../types";
+import { ThermostatCardConfig, TileCardConfig } from "./types";
+
+@customElement("hui-tile-card")
+export class HuiTileCard extends LitElement implements LovelaceCard {
+  public static async getConfigElement(): Promise<LovelaceCardEditor> {
+    await import("../editor/config-elements/hui-tile-card-editor");
+    return document.createElement("hui-tile-card-editor");
+  }
+
+  public static getStubConfig(
+    hass: HomeAssistant,
+    entities: string[],
+    entitiesFallback: string[]
+  ): TileCardConfig {
+    const includeDomains = ["sensor", "light", "switch"];
+    const maxEntities = 1;
+    const foundEntities = findEntities(
+      hass,
+      maxEntities,
+      entities,
+      entitiesFallback,
+      includeDomains
+    );
+
+    return {
+      type: "tile",
+      entity: foundEntities[0] || "",
+    };
+  }
+
+  @property({ attribute: false }) public hass?: HomeAssistant;
+
+  @state() private _config?: TileCardConfig;
+
+  public setConfig(config: ThermostatCardConfig): void {
+    if (!config.entity) {
+      throw new Error("Specify an entity");
+    }
+
+    const supportToggle =
+      config.entity && DOMAINS_TOGGLE.has(computeDomain(config.entity));
+
+    this._config = {
+      tap_action: {
+        action: "more-info",
+      },
+      icon_tap_action: {
+        action: supportToggle ? "toggle" : "more-info",
+      },
+      ...config,
+    };
+  }
+
+  public getCardSize(): number {
+    return 1;
+  }
+
+  private _handleAction(ev: ActionHandlerEvent) {
+    handleAction(this, this.hass!, this._config!, ev.detail.action!);
+  }
+
+  private _handleIconAction() {
+    const config = {
+      entity: this._config!.entity,
+      tap_action: this._config!.icon_tap_action,
+    };
+    handleAction(this, this.hass!, config, "tap");
+  }
+
+  render() {
+    if (!this._config || !this.hass) {
+      return html``;
+    }
+    const entityId = this._config.entity;
+    const entity = entityId ? this.hass.states[entityId] : undefined;
+
+    if (!entity) {
+      return html`
+        <hui-warning>
```

Comment:
I wonder if this will the cards where we can make this a bit nicer. Instead of big yellow blocks, can we start making this show the card as disabled/greyed out, and state be "Entity not found" ?

Reply:
![image](https://user-images.githubusercontent.com/5878303/195805324-d5c417ed-b6ff-4e6c-acf7-0e690960686e.png) Like this 🙂

Reply:
Yes that is great.